keps/2724-topology-aware-scheduling/README.md

```diff
@@ -663,11 +683,21 @@ For a given PodSet Kueue:
   level. Kueue starts the search from the specified level, but if the PodSet
   does not fit, then it tries higher levels in the hierarchy.
 
-Kueue packs pods on domains starting from the domains with the most free capacity. However, Kueue can operate in two modes when it comes to choosing the last domain if there is more than one capable of accommodating the remaining pods:
-- `MostAllocated` - Kueue chooses the domain with the least available resource that is capable of accommodating all the pods to mitigate resource fragmentation
-- `LeastAllocated` - Kueue chooses the domain that has the most available resources, providing better nodes utilization
+Kueue packs pods on domains with different algorithms, depending on the annotation and chosen profile:
+- `MostFreeCapacity` algorithm - Kueue selects as many domains as needed (if it meets user's requirement) starting from the one with the most free capacity;
+- `LeastFreeCapacity` algorithm - Kueue selects as many domains as needed (if it meets user's requirement) starting from the one with the least free capacity;
+- `BestFit` algorithm - Kueue selects as many domains as needed (if it meets user's requirement) starting from the one with the most free capacity, but tries to optimize the last domain to leave as few free resources as possible.
+
+Selection of the algorithm depends on TAS profiles expressed by feature gates, and PodSet's annotation:
+
+| featuregate/annotation                  | preferred         | required          | unconstrained     |
+| --------------------------------------- | ----------------- | ----------------- | ----------------- |
+| None                                    | BestFit           | BestFit           | BestFit           |
+| TASProfileMostFreeCapacity (deprecated) | MostFreeCapacity  | MostFreeCapacity  | MostFreeCapacity  |
+| TASProfileMixed                         | BestFit           | BestFit           | LeastFreeCapacity |
+| TASProfileLeastFreeCapacity             | LeastFreeCapacity | LeastFreeCapacity | LeastFreeCapacity |
```

Comment:
I would suggest to make these two feature gates deprecated too. The purpose of introducing them varies:
- `TASProfileMostFreeCapacity` - introduced as a bailout.
- `TASProfileMixed` - introduced for experimentation and collecting feedback
-  `TASProfileLeastFreeCapacity` - introduced for experimentation and collecting feedback

But I think all of them are "deprecated", because none of them is graduable in the current form.
I would suggest to:
- open an issue to introduce TAS configuration
- add a graduation criteria for TAS for beta to introduce the configuration

Reply:
Done